tests/BuildScriptGenerator.Tests/DefaultDockerfileGeneratorTest.cs

```diff
@@ -0,0 +1,236 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Castle.Core.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Oryx.BuildScriptGenerator.Exceptions;
+using Microsoft.Oryx.Tests.Common;
+using System.Xml;
+using Xunit;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Tests
+{
+    public class DefaultDockerfileGeneratorTest : IClassFixture<TestTempDirTestFixture>
+    {
+        private const string _buildImageFormat = "mcr.microsoft.com/oryx/build:{0}";
+        private const string _argRuntimeFormat = "ARG RUNTIME={0}:{1}";
+
+        private readonly string _tempDirRoot;
+
+        public DefaultDockerfileGeneratorTest(TestTempDirTestFixture testFixture)
+        {
+            _tempDirRoot = testFixture.RootDirPath;
+        }
+
+        [Fact]
+        public void GenerateDockerfile_Throws_IfNoPlatformIsCompatible()
+        {
+            // Arrange
+            var generator = CreateDefaultDockerfileGenerator(platforms: new IProgrammingPlatform[] { });
+            var ctx = CreateDockerfileContext(null, null);
+
+            // Act & Assert
+            var exception = Assert.Throws<UnsupportedLanguageException>(
+                () => generator.GenerateDockerfile(ctx));
+        }
+
+        [Theory]
+        [InlineData("dotnetcore", "2.0", "latest")]
+        [InlineData("dotnetcore", "2.1", "slim")]
+        [InlineData("dotnetcore", "3.0", "latest")]
+        [InlineData("node", "6.17", "latest")]
+        [InlineData("node", "8.16", "slim")]
+        [InlineData("node", "10.16", "slim")]
+        [InlineData("node", "12.12", "latest")]
+        [InlineData("php", "5.6", "latest")]
+        [InlineData("php", "7.3", "latest")]
+        [InlineData("python", "2.7", "latest")]
+        [InlineData("python", "3.7", "slim")]
+        [InlineData("python", "3.8", "latest")]
+        public void GenerateDockerfile_GeneratesBuildTagAndRuntime_ForProvidedPlatformAndVersion(
```

Comment:
what happens when the platform only has major version

Reply:
Since we format the version returned by the detector to include _at most_ the minor version, if a version is returned that is just a major version (either found by the detector or provided by the user), it will be respected and used in the Dockerfile template.